src/test/java/LoginFormTest.java

```
@@ -0,0 +1,115 @@
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
+        driver.get("https://magento.softwaretestingboard.com/customer/account/login/");
+    }
+
+    @Test
+    public void loginPage_LoginWithValidCredentials_Success() {
+        //given
+        LoginPage loginpage = new Homepage(driver).openLoginPage();
+
+        //when
+        Homepage homepage = loginpage.loginWithEmailAndPassword("ala.k@wp.pl", "Password!1");
+
+        //then
+        Assertions.assertThat(homepage.getWelcomeText()).isEqualTo("Welcome, Ala Kowalska!");
+    }
+
+    @Test
+    public void loginPage_LoginWithValidEmailAndInvalidPassword_FailureToLogInWithErrorMessageIncorrectSignInDisplayed(){
+        //given
+        String validEmail = "ala.k@wp.pl";
+        String invalidPassword = "XXXXX";
+
+        //when
+        LoginPage loginPage = new LoginPage(driver).loginWithInvalidEmailAndPassword(validEmail, invalidPassword);
+
+        //then
+        Assertions.assertThat(driver.findElement(By.cssSelector(".message-error.error.message")).getText()).
+                isEqualTo("The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.");
+    }
+    @Test
+    public void loginPage_LoginWithInvalidEmailAndInvalidPassword_FailureToLogInWithErrorMessageIncorrectSignInDisplayed(){
+        //given
+        String invalidEmail = "test@test.pl";
+        String invalidPassword = "XXXXX";
+
+        //when
+        LoginPage loginPage = new LoginPage(driver).loginWithInvalidEmailAndPassword(invalidEmail, invalidPassword);
+
+        //then
+        Assertions.assertThat(driver.findElement(By.cssSelector(".message-error.error.message")).getText()).
+                isEqualTo("The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.");
+    }
+
+    @Test
+    public void loginPage_LoginWithInvalidEmailAndEmptyPassword_FailureToLogInWithErrorMessageThisIsRequiredFieldDisplayed(){
```

`FailureWithCorrectFormErrorMessageDisplayed` instead of `FailureToLogInWithErrorMessageThisIsRequiredFieldDisplayed is enough for these and below tests.